custom_components/mammotion/sensor.py

```diff
@@ -186,4 +175,6 @@ def __init__(
     @property
     def native_value(self) -> StateType:
         """Return the state of the sensor."""
-        return self.entity_description.value_fn(self.coordinator.data)
+        return self.entity_description.value_fn(
+            self.coordinator.data.sys.toapp_report_data
```

Comment:
As much as I like this there is a ton of data under data.nav and the other large protobuf messages that we may want in sensors

Reply:
However we are not using it right now so rather the clean code for now.

Reply:
that's my thought exactly 😁